nvtabular/workflow/workflow.py

```diff
@@ -89,23 +90,32 @@ def transform(self, dataset: Union[Dataset, DataFrameType]) -> Union[Dataset, Da
 
         Parameters
         -----------
-        dataset: Dataset
-            Input dataset to transform
+        data: Union[Dataset, DataFrameType]
+            Input dataset or dataframe to transform
 
         Returns
         -------
-        Dataset
-            Transformed Dataset with the workflow graph applied to it
+        Dataset or DataFrame
+            Transformed Dataset or DataFrame with the workflow graph applied to it
         """
-        if isinstance(dataset, Dataset):
-            return self._transform_impl(dataset)
-        elif isinstance(dataset, pd.DataFrame) or (cudf and isinstance(dataset, cudf.DataFrame)):
-            return self._transform_df(dataset)
-        else:
-            raise ValueError(
-                "Workflow.transform received an unsupported type: {type(dataset)} "
-                "Supported types are a `merlin.io.Dataset` or DataFrame (pandas or cudf)"
-            )
+        raise NotImplementedError(
+            "Workflow.transform received an unsupported type: {type(dataset)} "
+            "Supported types are a `merlin.io.Dataset` or DataFrame (pandas or cudf)"
+        )
+
+    @transform.register
+    def _(self, dataset: Dataset) -> Dataset:
+        return self._transform_impl(dataset)
+
+    @transform.register
+    def _(self, dataframe: pd.DataFrame) -> pd.DataFrame:
+        return self._transform_df(dataframe)
+
+    if cudf:
```

Comment:
I'm guessing this is because `DataFrameType` doesn't work for dispatch? 👍🏻

Reply:
I didn't actually try it before making this commit I have now to check this and get the following error

```
nvtabular/workflow/workflow.py:112: in Workflow
    def _(self, dataframe: DataFrameType) -> DataFrameType:
/usr/lib/python3.8/functools.py:907: in register
    return self.dispatcher.register(cls, func=method)
/usr/lib/python3.8/functools.py:860: in register
    raise TypeError(
E   TypeError: Invalid annotation for 'dataframe'. typing.Union[pandas.core.frame.DataFrame, cudf.core.dataframe.DataFrame] is not a class.
```

Reply:
Yeah, that makes sense. I’m good with it as is, was just curious.